doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,152 @@
+   differ from this depending on best practices, etc.):
+    
+    ```
+    YYYYMMDDTHHMMSS.mmmm: X added asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X modified asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X deleted asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X modified dandiset metadata
+    ```
+    
+3. **Extensible significant action list.** The significant actions the system is
+   capable of tracking should be extensible--that is, in the future new actions
+   should be able to be added to the system, and existing actions should be able
+   to expand the specific data they carry.
+
+## Significant Actions
+
+The following is the list of actions performed on Dandisets that are considered
+"significant" (i.e., require an audit record to be generated), along with any
+additional, event-specific data each will carry with it.
+
+|  | General data | Specific data |
+| --- | --- | --- |
+| Create Dandiset | date_time, user_id, dandiset_id | metadata |
+| Add owner | date_time, user_id, dandiset_id | owner_id |
+| Remove owner | date_time, user_id, dandiset_id | owner_id |
+| Update Dandiset metadata | date_time, user_id, dandiset_id | metadata |
+| Add asset | date_time, user_id, dandiset_id | asset_path, checksum, metadata, blob_id |
+| Update asset | date_time, user_id, dandiset_id | asset_path, checksum, blob_id, metadata |
+| Remove asset | date_time, user_id, dandiset_id | asset_path |
+| Publish Dandiset | date_time, user_id, dandiset_id | version |
+| Delete Dandiset | date_time, user_id, dandiset_id | none |
+
+## Asset chain analysis
+
+This section provides some analysis of the historical data living in the
+"`Asset` chains" kept in the `Asset.previous` links. The immediate goal is to
+determine how much "significant action" data can be reconstructed therefrom;
+there is also some analysis of how this may affect provenance work in the
+future.
+
+For context: because most changes to an `Asset` model cause a new model instance
+to be created, and because that new `Asset` instance points to its "ancestor"
+via a `previous` property, the history of an asset is at least partially
+traceable through this "asset chain".
+
+### Audit
+
+| Action | Recoverable fields | Unrecoverable fields | Notes |
+| --- | --- | --- | --- |
+| Create Dandiset | date_time, dandiset_id | user_id, metadata | date_time is equal to the Dandiset’s created_at value. user_id and metadata (as they were at the beginning of the Dandiset’s lifecycle) cannot be known with certainty.<br><br> Dandisets that have already been deleted will not have any recoverable fields (though we believe it very unlikely that a Dandiset has ever actually been deleted).<br><br> Independent of asset chains. |
+| Add owner | none | date_time, user_id, dandiset_id, owner_id | These events are all "summed" into the current owners list, with no way to know the evolution thereof.<br><br> Independent of asset chains. |
```

Comment:
Is this indicating that we wouldn't know the time at which an owner is added? Why is that?